### PR TITLE
use javaparser for codegen

### PR DESCRIPTION
**Pull request by evacchi**
hey @mswiderski  here's an example of usage of JavaParser. the test you wrote dumps reasonable code, only negative part is the method chain is on one single line, but I think the pretty printer can be configured (will ping the guys from JP for hints)

BTW please take a look at this code, and then feel free to ping me. As you can see it's a bit more verbose, but it's easy to wrap things inside of helpers, like we did for e.g. DMN. 

The Mojo part I didn't have time to update, but it's pretty much the same thing :)

### CODE REVIEW

File: jbpm/jbpm-flow/pom.xml [hunk 1]
```diff
@@ -29,16 +29,20 @@
       <groupId>org.kie</groupId>
       <artifactId>kie-internal</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.drools</groupId>
+      <artifactId>drlx-parser</artifactId>
```
Comment (mswiderski) [inline jbpm/jbpm-flow/pom.xml:34]: I don't think we should have it in the jbpm-flow, looks like we need to separate/interface the generation part from pure fluent api which is out exec model

**Review by mswiderski — COMMENTED** [review]
thanks @evacchi, this makes it more aligned with code gen instead of string concat :)

one issue is we need to move out the code gen from the jbpm-flow which is purely runtime dependency so we should not have anything that is compile time

**Comment by evacchi** [comment]
> one issue is we need to move out the code gen from the jbpm-flow which is purely runtime dependency so we should not have anything that is compile time

100% agree, just wanted to show you some code. We can certainly think of a better way to do this.

File: jbpm/jbpm-flow/pom.xml (hunk 1 above)
Comment (evacchi) [reply jbpm/jbpm-flow/pom.xml:34]: agreed

**Pull request closed without merging**